From b8007e1cc6d9423b3fe8a39df24f1816617c38f0 Mon Sep 17 00:00:00 2001
From: Ramneek Mehresh <ramneek.mehresh@freescale.com>
Date: Mon, 23 Mar 2015 18:19:52 +0530
Subject: [PATCH 100/124] powerpc:rcpm:usb: Fix kernel crash in setting usb as
 wakeup source

Fix kernel crash issue in setting usb as wake-up source for system
sleep feature for relevant qoriq platforms (those supporting system
sleep via rcpm)

Signed-off-by: Ramneek Mehresh <ramneek.mehresh@freescale.com>
Change-Id: I838234aff00800b6fafbb02ec12de7d3f2cc6075
Reviewed-on: http://git.am.freescale.net:8181/33283
Tested-by: Review Code-CDREVIEW <CDREVIEW@freescale.com>
Reviewed-by: Honghua Yin <Hong-Hua.Yin@freescale.com>
---
 drivers/usb/core/usb.c           | 1 -
 drivers/usb/host/fsl-mph-dr-of.c | 3 ---
 2 files changed, 4 deletions(-)

diff --git a/drivers/usb/core/usb.c b/drivers/usb/core/usb.c
index 4d8cd50..8d5b2f4 100644
--- a/drivers/usb/core/usb.c
+++ b/drivers/usb/core/usb.c
@@ -445,7 +445,6 @@ struct usb_device *usb_alloc_dev(struct usb_device *parent,
 	dev->dev.groups = usb_device_groups;
 	dev->dev.dma_mask = bus->controller->dma_mask;
 	set_dev_node(&dev->dev, dev_to_node(bus->controller));
-	dev->dev.of_node = bus->controller->of_node;
 	dev->state = USB_STATE_ATTACHED;
 	dev->lpm_disable_count = 1;
 	atomic_set(&dev->urbnum, 0);
diff --git a/drivers/usb/host/fsl-mph-dr-of.c b/drivers/usb/host/fsl-mph-dr-of.c
index 79d03e4..0573c8b 100644
--- a/drivers/usb/host/fsl-mph-dr-of.c
+++ b/drivers/usb/host/fsl-mph-dr-of.c
@@ -103,9 +103,6 @@ static struct platform_device *fsl_usb2_device_register(
 	else
 		dma_set_mask(&pdev->dev, DMA_BIT_MASK(32));
 
-	if (pdata->operating_mode != FSL_USB2_DR_DEVICE)
-		pdev->dev.of_node = ofdev->dev.of_node;
-
 	retval = platform_device_add_data(pdev, pdata, sizeof(*pdata));
 	if (retval)
 		goto error;
-- 
2.1.0.27.g96db324

